drivers/platform/stm32/stm32_spi.c

```diff
@@ -288,14 +292,24 @@ int32_t stm32_spi_transfer(struct no_os_spi_desc *desc,
 			no_os_udelay(msgs[i].cs_delay_first);
 
 		__HAL_SPI_ENABLE(&sdesc->hspi);
-		while(msgs[i].bytes_number--) {
+
+		while ((msgs[i].rx_buff && rx_cnt < msgs[i].bytes_number) ||
+		       (msgs[i].tx_buff && tx_cnt < msgs[i].bytes_number)) {
 			while(!(SPIx->SR & SPI_SR_TXE))
 				;
-			*(volatile uint8_t *)&SPIx->DR = *msgs[i].tx_buff++;
+			if (msgs[i].tx_buff)
+				*(volatile uint8_t *)&SPIx->DR = msgs[i].tx_buff[tx_cnt++];
```

Comment:
ok, for stm32 I understand what you did, HAL is not used for transfer so we need to take care of the pointers, but the question still remains for pico platform

Reply:
the same goes for pico platform, please see the code from: https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_spi/spi.c#L76
dst and src are not checked if they are ```NULL```